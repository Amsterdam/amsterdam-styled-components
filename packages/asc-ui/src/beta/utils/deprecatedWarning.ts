/**
 * A general deprecated console warning avoided by lint
 */

export default (message: string) => {
  // eslint-disable-next-line
  console.warn(message)
}
