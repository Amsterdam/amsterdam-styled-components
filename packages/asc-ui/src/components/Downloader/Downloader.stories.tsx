import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import Downloader from './Downloader'

const DownloaderComponent: React.FC<{}> = () => (
  <>
    <Downloader
      imageSrc="https://acc.cms.data.amsterdam.nl/sites/default/files/images/2019-factsheet-jeugdwerkloosheid-2014-2018.png"
      onClick={action('click')}
      description="Download PDF (12MB)"
    ></Downloader>
  </>
)

storiesOf('Atoms/Downloader', module)
  .addDecorator(storyFn => (
    <div style={{ padding: '40px 10px' }}>{storyFn()}</div>
  ))
  .add('default state', () => <DownloaderComponent />)
