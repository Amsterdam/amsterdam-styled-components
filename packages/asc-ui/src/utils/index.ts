export { default as fromTheme, valueFromObject } from './fromTheme'
export {
  color,
  getColorCode,
  getTypographyFromTheme,
  focusStyleOutline,
  srOnlyStyle,
  breakpoint,
  svgFill,
  showHide,
  ShowHideTypes,
  customCss,
  CustomCssPropsType,
  CustomCssPropsInterface,
} from './themeUtils'
