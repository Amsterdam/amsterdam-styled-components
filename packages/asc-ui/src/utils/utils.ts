import { AbstractNode, IconDefinition } from '@datapunt/asc-icons/lib/types'
import * as React from 'react'

export function log(message: string) {
  if (!(process && process.env && process.env.NODE_ENV === 'production')) {
    console.error(`[icons-react]: ${message}.`)
  }
}

export function isIconDefinition(target: any): target is IconDefinition {
  return (
    typeof target === 'object' &&
    typeof target.name === 'string' &&
    typeof target.icon === 'object'
  )
}

export function normalizeAttrs(attrs: Attrs = {}): Attrs {
  return Object.keys(attrs).reduce((acc: Attrs, key) => {
    const val = attrs[key]
    switch (key) {
      case 'class':
        acc.className = val
        delete acc.class
        break
      default:
        acc[key] = val
    }
    return acc
  }, {})
}

export interface Attrs {
  [key: string]: string
}

export class MiniMap<V> {
  get size() {
    return Object.keys(this.collection).length
  }

  private collection: { [key: string]: V } = {}

  clear(): void {
    this.collection = {}
  }

  delete(key: string): boolean {
    return delete this.collection[key]
  }

  get(key: string): V | undefined {
    return this.collection[key]
  }

  has(key: string): boolean {
    return Boolean(this.collection[key])
  }

  set(key: string, value: V): this {
    this.collection[key] = value
    return this
  }
}

export function generate(
  node: AbstractNode,
  key: string,
  rootProps?: { [key: string]: any } | false,
): any {
  if (!rootProps) {
    return React.createElement(
      node.tag,
      { key, ...normalizeAttrs(node.attrs) },
      (node.children || []).map((child, index) =>
        generate(child, `${key}-${node.tag}-${index}`),
      ),
    )
  }
  return React.createElement(
    node.tag,
    {
      key,
      ...normalizeAttrs(node.attrs),
      ...rootProps,
    },
    (node.children || []).map((child, index) =>
      generate(child, `${key}-${node.tag}-${index}`),
    ),
  )
}
