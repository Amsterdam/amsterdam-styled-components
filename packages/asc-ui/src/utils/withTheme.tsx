import * as React from 'react'
import { shallow, render, mount } from 'enzyme'
import { ThemeProvider } from '../index'

export const renderWithTheme = (component: JSX.Element) =>
  render(<ThemeProvider>{component}</ThemeProvider>)

export const shallowWithTheme = (component: JSX.Element) =>
  shallow(<ThemeProvider>{component}</ThemeProvider>)

export const mountWithTheme = (component: JSX.Element) =>
  mount(<ThemeProvider>{component}</ThemeProvider>)
