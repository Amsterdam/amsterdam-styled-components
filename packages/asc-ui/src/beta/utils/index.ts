import deprecatedWarning from './deprecatedWarning'

export {
  themeColor,
  srOnlyStyle,
  breakpoint,
  svgFill,
  fluidTypoStyle,
  typographyStyle,
  showAboveBackDrop,
  getValueFromTheme,
  themeSpacing,
} from './themeUtils'
export { valueFromObject } from './valueFromObject'
export { fromProps } from './fromProps'

export { deprecatedWarning }
