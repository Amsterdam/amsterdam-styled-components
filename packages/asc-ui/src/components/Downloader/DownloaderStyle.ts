import styled from '@datapunt/asc-core'
import { getColorFromTheme, breakpoint } from '../../utils'

const DownloaderStyle = styled.div<{}>`
  width: 100%;

  @media screen and ${breakpoint('max-width', 'tabletM')} {
    margin-bottom: 20px;
  }

  & > div:first-child {
    background-color: ${({ theme }) =>
      getColorFromTheme(theme, 'tint', 'level2')};
  }
`

export default DownloaderStyle
