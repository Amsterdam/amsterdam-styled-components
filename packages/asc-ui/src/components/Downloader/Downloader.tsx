import React from 'react'
import { Download } from '@datapunt/asc-assets'
import DownloaderStyle, { DownloaderStyleProps } from './DownloaderStyle'
import { Row, Column } from '../Grid'
import Button from '../Button'
import Image from '../Image'
import DownloaderContent from './DownloaderContent'

interface Props extends DownloaderStyleProps {
  imageSrc: string
  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
  description: string
}

const Downloader: React.FC<Props> = ({
  imageSrc,
  onClick,
  description,
  ...otherProps
}) => (
  <DownloaderStyle {...otherProps}>
    <Row>
      <Column span={{ small: 1, medium: 4, big: 6, large: 12, xLarge: 12 }}>
        <Column
          span={{ small: 1, medium: 2, big: 4, large: 10, xLarge: 10 }}
          push={{ small: 0, medium: 1, big: 1, large: 1, xLarge: 1 }}
        >
          <DownloaderContent>
            <Image src={imageSrc} alt=""></Image>
            <Button color="primary" onClick={onClick}>
              <Download />
              {description}
            </Button>
          </DownloaderContent>
        </Column>
      </Column>
    </Row>
  </DownloaderStyle>
)

export default Downloader
