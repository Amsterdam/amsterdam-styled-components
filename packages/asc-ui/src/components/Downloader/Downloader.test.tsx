import * as React from 'react'
import Downloader from './Downloader'
import { renderWithTheme } from '../../utils/withTheme'
import 'jest-styled-components'

jest.useFakeTimers()

describe('TopBar', () => {
  let component: Cheerio

  beforeEach(() => {
    component = renderWithTheme(
      <Downloader
        backgroundColor="red"
        imageSrc="https://data.amsterdam.nl/assets/images/amsterdam-maps.png"
        onClick={() => {}}
        description="Download PDF (12MB)"
      ></Downloader>,
    )
  })

  it('should render', () => {
    expect(component).toMatchSnapshot()
  })
})
