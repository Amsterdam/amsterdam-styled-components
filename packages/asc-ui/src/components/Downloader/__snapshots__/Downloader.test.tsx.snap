// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Downloader should render 1`] = `
.c0 {
  width: 100%;
  background-color: red;
}

.c2 {
  box-sizing: border-box;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c1 {
  box-sizing: border-box;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  max-width: 1430px;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 .layout-label {
  position: absolute;
  top: -20px;
  width: 100%;
}

.c1 .layout-label::before {
  position: absolute;
  padding: 2px 4px;
  font: 12px sans-serif;
  background: white;
  height: 20px;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  -webkit-transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  background: #004699;
  padding: 0 10px 0 10px;
  min-height: 38px;
  line-height: 1em;
  color: #fff;
}

.c6:focus {
  z-index: 10;
  outline-color: #FEC813;
  outline-style: solid;
  outline-offset: 0px;
  outline-width: 3px;
}

.c6:hover {
  background: #00387a;
}

.c6:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c6 > svg {
  width: 30px;
  height: 30px;
}

.c6 > svg rect,
.c6 > svg polygon,
.c6 > svg path {
  fill: #ffffff;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 32px auto;
}

.c3 > .c4 {
  margin-bottom: 32px;
  width: 100%;
}

.c3 > .c5 > svg {
  width: 20px;
  height: 20px;
}

@media (min-width:1200px) {
  .c2 {
    max-width: 100%;
  }
}

@media (min-width:1024px) and (max-width:1199px) {
  .c2 {
    max-width: 100%;
  }
}

@media (min-width:768px) and (max-width:1023px) {
  .c2 {
    max-width: 100%;
  }
}

@media (min-width:414px) and (max-width:767px) {
  .c2 {
    max-width: 100%;
  }
}

@media (max-width:413px) {
  .c2 {
    max-width: 100%;
  }
}

@media (min-width:1200px) {
  .c1 {
    padding-left: 44px;
    padding-right: 44px;
  }
}

@media (min-width:1024px) and (max-width:1199px) {
  .c1 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

@media (min-width:768px) and (max-width:1023px) {
  .c1 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:414px) and (max-width:767px) {
  .c1 {
    padding-left: 20px;
    padding-right: 20px;
  }
}

@media (max-width:413px) {
  .c1 {
    padding-left: 20px;
    padding-right: 20px;
  }
}

@media screen and (max-width:540px) {
  .c3 {
    margin: 20px auto;
  }
}

@media screen and (max-width:540px) {
  .c3 > .c4 {
    margin-bottom: 20px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      order="0"
    >
      <div
        class="c3"
      >
        <img
          alt=""
          class="c4 "
          src="https://data.amsterdam.nl/assets/images/amsterdam-maps.png"
        />
        <button
          class="c5 c6"
          color="primary"
        >
          <svg
            focusable="false"
            viewBox="0 0 32 32"
          >
            <path
              d="M0 27.996h32V32H0v-4.004zm22.36-13.983l2.83 2.823L16 26.044l-9.19-9.208 2.83-2.823L14 18.377V0h4v18.377l4.36-4.364z"
            />
          </svg>
          Download PDF (12MB)
        </button>
      </div>
    </div>
  </div>
</div>
`;
